controlplane/internal/controllers/rke2controlplane_controller.go

```diff
@@ -346,7 +346,8 @@ func (r *RKE2ControlPlaneReconciler) updateStatus(ctx context.Context, rcp *cont
 		return fmt.Errorf("no Control Plane Machines are ready for RKE2ControlPlane %s/%s", rcp.Namespace, rcp.Name)
 	}
 
-	availableCPMachines := readyMachines.Filter(collections.Not(collections.HasUnhealthyCondition))
+	availableCPMachines := readyMachines
+	//.Filter(collections.Not(collections.HasUnhealthyCondition))
```

Comment:
Well, this is part of an improvement that I mentioned previously to @alexander-demicev and @richardcase. If we only select Healthy machines, we constraint ourself too much because of the way CAPI defines a healthy machine.

For instance: on AWS, if you deploy the external Cloud Provider, the machine will only be healthy if the cloud provider populates the `providerID` field in the workload cluster's node. If, for example, your Cloud Provider does not deploy on your first node because your first node has a `ControlPlane` taint, the machine will never be healthy, and will therefore never be added to the list of available IPs for registration of other nodes.

RKE2 does not need this kind of behaviour. You can make additional nodes register with the init nodes when these node is `Ready`, not necessarily when it is healthy and with a valid `ProviderID` field.

Reply:
well the cp taint is users responsibility to put or leave out right

Reply:
Thanks for putting together the detailed explanation, it generally makes sense. Please remove the leftover commented code